src/fileio.cpp

```diff
@@ -51,85 +50,99 @@ const char* LoadingError::what() const noexcept
+            entries.push_back(entryName);
+        }
+    }
+    return entries;
+}
 
-/**
- * @fn void DirectoryReader::ClosePath()
- * Denote no further interest in reading the contents of the current directory.
- */
 
 /**
- * @fn const char *DirectoryReader::MakePath(const char *directory, const char *fname)
- * Construct a path from a directory and a file name.
- * @param directory Directory part of the path.
- * @param fname File name part of the path.
- * @return The combined path to the file.
+ * Delete the given file.
+ * @param path Path to delete.
  */
+void RemoveFile(const std::string &path)
+{
 
-/**
- * @fn bool DirectoryReader::EntryIsFile()
- * Test whether the last returned entry from #NextEntry is a file.
- * @return Whether the entry is a file.
- */
+	if (!PathIsFile(path)) return;
+	try {
+	    std::filesystem::remove(path);
+	} catch(std::filesystem::filesystem_error const& ex) {
+	        error("%s", ex.what());
+	}
 
-/**
- * @fn bool DirectoryReader::EntryIsDirectory()
- * Test whether the last returned entry from #NextEntry is a directory.
- * @return Whether the entry is a directory.
- */
+}
 
 /**
- * Get the next file entry.
- * Pulls entries from #NextEntry, until the end is reached or until a file is returned.
- * @return Next file entry, if available.
+ * Test whether the given path points to a normal file.
+ * @param path Path to investigate.
+ * @return Whether the given path points to a normal file.
  */
-const char *DirectoryReader::NextFile()
+bool PathIsFile(const std::string &path)
 {
-	const char *entry;
+	    std::filesystem::path fs(path);
+        try {
+	        return std::filesystem::exists(fs) && std::filesystem::is_regular_file(fs);
+	    } catch(std::filesystem::filesystem_error const& ex) {
+	        error("%s", ex.what());
+	        return false;
+	    }
 
-	do {
-		entry = this->NextEntry();
-	} while (entry != nullptr && !this->EntryIsFile());
-	return entry;
 }
 
-
 /**
- * Construct a directory reader object (specific for the operating system).
- * @return A directory reader.
- * @ingroup fileio_group
+ * Test whether the given path points to a directory.
+ * @param path Path to investigate.
+ * @return Whether the given path points to a directory.
  */
-DirectoryReader *MakeDirectoryReader()
+bool PathIsDirectory(const std::string &path)
 {
-#ifdef LINUX
-	return new UnixDirectoryReader();
-#elif WINDOWS
-	return new WindowsDirectoryReader();
-#else
-	assert_compile(false);
-#endif
+	    std::filesystem::path fs(path);
+
+	    try {
+	        return std::filesystem::exists(fs) && std::filesystem::is_directory(fs);
+	    } catch(std::filesystem::filesystem_error const& ex) {
+	        error("%s", ex.what());
+	        return false;
+	    }
 }
 
 /**
  * RCD file reader constructor, loading data from a file.
  * @param fname Name of the file to load.
  */
-RcdFileReader::RcdFileReader(const char *fname)
+RcdFileReader::RcdFileReader(std::string fname)
```

Comment:
```suggestion
RcdFileReader::RcdFileReader(const std::string& fname)
```
I think this should be fine here.